src/neuroconv/basedatainterface.py

```diff
@@ -82,6 +86,11 @@ def run_conversion(
                 " use DataInterface.add_to_nwbfile()."
             )
 
+        base_metadata = self.get_metadata()
+        if metadata is None:
+            metadata = {}
+        metadata = dict_deep_update(base_metadata, metadata)
```

Comment:
@luiztauffer thanks, I agree this is a bug. We should either make `metadata` required or properly handle when it is missing. Is there a situation where `self.get_metadata()` returns `None`? Otherwise, let's just simplify this as
```suggestion
        metadata = dict_deep_update(self.get_metadata(), metadata)
```

Reply:
@bendichter 
>  Is there a situation where self.get_metadata() returns None?

No, but `metadata` might be `None`, that's why we need lines 90-91

Reply:
oh, right I see